syntax = "proto3";

import "shiva/core/communication_objects/specs.proto";

package communication_objects;

enum ComponentType {
    META = 0;
    LEARNER = 1;
    MULTIENV = 2;
    ENVIRONMENT = 3;
    EVALUATION = 4;
    AGENTS = 5;
}

enum StatusType {
    ERROR = 0;
    // module is still "launching" or waiting something
    LAUNCHING = 1;
    // ready to run or running
    RUN = 2;
}

// Message used for components to Check-In and send initial parameters in between modules
message StatusProto {
    int32 id = 1;
    ComponentType type = 2;
    StatusType status = 3;
    string address = 4;
    oneof specs {
        EnvSpecsProto env = 5;
        MultiEnvSpecsProto menv = 6;
        LearnerSpecsProto learner = 7;
    }
    string extra = 8;
}

enum LoadType {
    LOCAL = 0;
    AWS = 1;
}

message ConfigProto {
    ComponentType type = 1;
    LoadType load_type = 2;
    string load_path = 3;
}
